examples/Azure_IoT_Hub_ESP32/AzIoTSasToken.cpp

```diff
@@ -0,0 +1,282 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "AzIoTSasToken.h"
+#include "SerialLogger.h"
+#include <az_result.h>
+#include <mbedtls/base64.h>
+#include <mbedtls/md.h>
+#include <mbedtls/sha256.h>
+#include <stdlib.h>
+#include <time.h>
+
+#define INDEFINITE_TIME ((time_t)-1)
+
+#define az_span_is_empty(x) (az_span_size(x) == az_span_size(AZ_SPAN_EMPTY) && az_span_ptr(x) == az_span_ptr(AZ_SPAN_EMPTY))
+
+static uint32_t getSasTokenExpiration(const char* sasToken)
```

Comment:
We wouldn't consider modifying the samples as a breaking change, correct?

In that case, when would we version increment this repo from 1.0.0?

Reply:
AFAIK, we can modify the samples. That will not be a problem for our users on Arduino land. But we also do not expect major breaking changes.

We can change and increment the version of this library whenever we need to.